Quotient/events/roommessageevent.cpp

```diff
@@ -219,6 +219,49 @@ QString RoomMessageEvent::replacedBy() const
     // clang-format on
 }
 
+bool RoomMessageEvent::isReply() const
+{
+    const auto relations = contentPart<QJsonObject>("m.relates_to"_ls);
+    if (!relations.isEmpty()) {
+        const bool hasReplyRelation = relations.contains("m.in_reply_to"_ls);
+        bool isFallingBack = relations["is_falling_back"_ls].toBool();
+        return hasReplyRelation && !isFallingBack;
+    }
+    return false;
```

Comment:
Add `bool isFallingBack;` to `EventRelation`, add it to `JsonConverter<EventRelation>`, and then the following can be here:
```suggestion
    const auto relation = getUpstreamRelation();
    return relation && relation.type == EventRelation::ReplyType && !relation.isFallingBack;
```